### ISSUE TITLE
x2many fields glitch when switching between records

### ISSUE THREAD

**Issue by NobodyIII** [odoo/odoo#17590]
Impacted versions: 9.0

Steps to reproduce: 
1. Find an editable x2many table, e.g. in Sales module.
2. Switch to edit mode.
3. Without modifying anything else, click into one x2many entry and modify a single field.
4. Without clicking anything else, switch to a different record by clicking one of the arrows on the upper pane, by the view type buttons.

Current behavior:
After step 4, one of the x2many table rows is carried over and overlaid on top of the x2many table for the new record.

Expected behavior:
Odoo should switch records cleanly, rather than carrying over a single table row and placing it on top of the table for the newly selected record. Ideally, it should also ask whether you want to save the change you made in the table.

Video/Screenshot link (optional):
![image](https://user-images.githubusercontent.com/10506218/27105652-98c719be-504d-11e7-96e4-bd618f017728.png)

**Comment by pedrobaeza** [comment]
I'm afraid this is something that can't be easily fixed without having to add a lot of code that can be risky with a very low benefit ratio. Any way, I can tell you that views have been slightly improved in v10, and rewritten on v11, and this weird effect has been mitigated in most parts, so we can consider this as "won't fix" for version 9, which is also at the end of its life cycle, but mostly solved on upper versions.

**Disclaimer**: I'm not an Odoo employee, but a community member that voluntary contributes to triage / classify / solve issues. Take also into account that there's no obligation to answer nor solving things put in Odoo Community issue tracker. And thus, please consider this as a best effort initiative.
